bin/geni-sync-wireless

```diff
@@ -0,0 +1,618 @@
+        self.PROJECT_ATTRIBUTE_TABLE = Table('pa_project_attribute', 
+                                             self._metadata, autoload=True)
+        self.PROJECT_MEMBER_TABLE = Table('pa_project_member', 
+                                          self._metadata, autoload=True)
+        self.MEMBER_ATTRIBUTE_TABLE = Table('ma_member_attribute',
+                                          self._metadata, autoload=True)
+        self.SSH_KEY_TABLE = Table('ma_ssh_key',
+                                   self._metadata, autoload=True)
+        self.SERVICE_REGISTRY_TABLE = Table('service_registry', 
+                                   self._metadata, autoload=True)
+
+        self.holdingpen_group_description = "GENI ORBIT MEMBER HOLDINGPEN"
+
+        self._base_orbit_url = self.get_orbit_base_url()
+        self._orb = ORBIT_Interface(self._base_orbit_url)
+
+        # These are instance variables filled in during synchronize
+
+        # GENI wimax-enabled projects and members of these projects
+        self._geni_projects = {}
+        self._geni_members = {}
+
+        # ORBIT groups and users
+        self._orbit_groups = {}
+        self._orbit_users = {}
+
+        # These keep state of deleted information in the current
+        # synchronize call between sub-methods
+        self._deleted_orbit_groups = []
+        self._deleted_orbit_members = {}
+
+    # Return URL for ORBIT Delegated AM REST API from service registry
+    def get_orbit_base_url(self):
+        WIMAX_SITE = 10
+        query = self._session.query(self.SERVICE_REGISTRY_TABLE)
+        query = query.filter(self.SERVICE_REGISTRY_TABLE.c.service_type == WIMAX_SITE)
+        rows = query.all()
+        if len(rows) == 0:
+            self.error("NO WIMAX_SITE (%s) defined in SERVICE_REGISTRY" % \
+                           WIMAX_SITE)
+        orbit_base_url = rows[0].service_url
+        return orbit_base_url
+
+    # Parse settings.php file and extract db_dsn value
+    def find_database_url(self):
+        data = None
+        try:
+            data = open(self._options.settings_file, 'r').read()
+        except Exception:
+            self.error("Error reading settings file: %s" % \
+                           self._options.settings_file)
+        lines = data.split('\n')
+        for line in lines:
+            if line.find("$db_dsn") == 0:
+                db_url = line.split('\'')[1].replace('pgsql', 'postgresql')
+                return db_url
+        self.error("No $db_dsn entry in settings file")
+        
+    # Print error and exit
+    def error(self, msg): print msg; sys.exit()
+
+    # Get pretty name from member data
+    def get_pretty_name(self, member_info):
+        if 'displayName' in member_info:
+            return member_info['displayName']
+        else:
+            return "%s %s" % (member_info['first_name'], 
+                              member_info['last_name'])
```

Comment:
Users don't always have both first_name and last_name. If both don't exist, fall back on the email address.